src/main/java/net/citizensnpcs/api/npc/NPC.java

```diff
@@ -154,8 +154,31 @@
      * @param trait
      *            Trait to get
      * @return Trait with the given name
+     * 
+     * @deprecated for intransparent naming. Use {@link #getOrAddTrait(Class)} for the same behavior. 
      */
+    @Deprecated
     public <T extends Trait> T getTrait(Class<T> trait);
+    
+    /**
+     * Gets a trait from the given class. If the NPC does not currently have the trait then it will be created and
+     * attached using {@link #addTrait(Class)} .
+     *
+     * @param trait
+     *            Trait to get
+     * @return Trait with the given name
+     */
+    public <T extends Trait> T getOrAddTrait(Class<T> trait);
+    
+    /**
+     * Gets a trait from the given class. If the NPC does not currently have the trait, 
+     * <code>null</code> will be returned.
+     *
+     * @param trait
+     *            Trait to get
+     * @return Trait with the given name
+     */
+    public <T extends Trait> T getTraitSafely(Class<T> trait);
```

Comment:
I'm not sold on getTraitSafely because there's no real extra safety here. Maybe you could call it getTraitOptional and return an Optional?

Reply:
I thought about that at first. But I (personally) don't really like working with optionals. Unless you completely set on functional programming, optionals just add boiler plate to your code, as you have to "unpack" them every time.

Reply:
I agree, but the name should be changed to something like getTraitNullable. Even using .computeTrait and .getTraitNullable might be in line with other newer Collections APIs.

Reply:
Just added a new commit and renamed the function